File: tests/typescript-zod/validation_constraints/output.ts
import { z } from "zod";

export const UserSchema = z.object({
  age: z.number().int().min(0).max(150),
  email: z.string().email(),
  rating: z.number().gt(0).lt(5).optional(),
  score: z.number().min(0).max(100).multipleOf(0.5).optional(),
  tags: z.array(z.string()).min(1).max(10).optional(),
  username: z.string().min(3).max(20).regex(/^[a-z_][a-z0-9_]*$/),
});
export type User = z.infer<typeof UserSchema>;

File: tests/python/validation_constraints/output.py
from __future__ import annotations

from typing import List
from pydantic import BaseModel, ConfigDict, EmailStr, Field




class User(BaseModel):
    model_config = ConfigDict(extra="forbid")

    age: int = Field(ge=0, le=150)
    email: EmailStr
    rating: float | None = Field(gt=0, lt=5, default=None)
    score: float | None = Field(ge=0, le=100, multiple_of=0.5, default=None)
    tags: List[str] | None = Field(min_length=1, max_length=10, default=None)
    username: str = Field(min_length=3, max_length=20, pattern=r"^[a-z_][a-z0-9_]*$")

